.github/workflows/macos.yml

```diff
@@ -16,17 +16,32 @@ jobs:
         brew update
         brew install cmake
         brew install libomp
+        brew list libomp
         brew install boost
         set -e
     - name: build
       run: |
         mkdir build && cd build
-        cmake .. -DCMAKE_INSTALL_PREFIX=/Users/runner/work/hipSYCL/hipSYCL/build/install
+        cmake .. \
+        -DOpenMP_libomp_LIBRARY=/usr/local/Cellar/libomp/15.0.5/lib/libomp.a \
```

Comment:
Obviously, just try to get it going as a first step, but in the longer run, you probably will want to detect this path automatically, as it will change with every LLVM update.
One example that might work: `OMP_ROOT= brew list libomp | grep libomp.a | sed -E "s/^(.*)/lib/libomp.a$/\1/"` and then use `$OMP_ROOT/lib/libomp.a` for the CMake invocation.